// material-ui-next-responsive-table/lib/DataTable.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Table = require('@material-ui/core/Table');

var _Table2 = _interopRequireDefault(_Table);

var _TableBody = require('@material-ui/core/TableBody');

var _TableBody2 = _interopRequireDefault(_TableBody);

var _TableCell = require('@material-ui/core/TableCell');

var _TableCell2 = _interopRequireDefault(_TableCell);

var _TableFooter = require('@material-ui/core/TableFooter');

var _TableFooter2 = _interopRequireDefault(_TableFooter);

var _TableHead = require('@material-ui/core/TableHead');

var _TableHead2 = _interopRequireDefault(_TableHead);

var _TableRow = require('@material-ui/core/TableRow');

var _TableRow2 = _interopRequireDefault(_TableRow);

var _Renderer = require('./Renderer');

var _NoContent = require('./NoContent');

var _NoContent2 = _interopRequireDefault(_NoContent);

var _Pagination = require('./Pagination');

var _Pagination2 = _interopRequireDefault(_Pagination);

var _lodash = require('lodash.isequal');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * Simple read only table with header and body
 */
var DataTable = function (_Component) {
  _inherits(DataTable, _Component);

  function DataTable() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, DataTable);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = DataTable.__proto__ || Object.getPrototypeOf(DataTable)).call.apply(_ref, [this].concat(args))), _this), _this.handleChangePage = function (event, page) {
      return _this.props.onChangePage(event, page);
    }, _this.getRowClass = function (index) {
      var rowsClassArray = _this.props.rowsClassArray;

      return rowsClassArray && rowsClassArray[index] ? rowsClassArray[index] : '';
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(DataTable, [{
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps) {
      var _props = this.props,
          enableShouldComponentUpdate = _props.enableShouldComponentUpdate,
          data = _props.data;

      if (enableShouldComponentUpdate) {
        return !(0, _lodash2.default)(nextProps.data, data);
      }
      return true;
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props2 = this.props,
          columns = _props2.columns,
          count = _props2.count,
          data = _props2.data,
          noContentText = _props2.noContentText,
          page = _props2.page,
          rowsPerPage = _props2.rowsPerPage,
          showPagination = _props2.showPagination,
          TableBodyCellProps = _props2.TableBodyCellProps,
          TableBodyProps = _props2.TableBodyProps,
          TableBodyRowProps = _props2.TableBodyRowProps,
          TableHeadCellProps = _props2.TableHeadCellProps,
          TableHeadProps = _props2.TableHeadProps,
          TableHeadRowProps = _props2.TableHeadRowProps,
          TablePaginationProps = _props2.TablePaginationProps,
          TableProps = _props2.TableProps;


      if (!Array.isArray(data) || data.length === 0 || !Array.isArray(columns) || columns.length === 0) {
        return _react2.default.createElement(_NoContent2.default, { text: noContentText });
      }

      return _react2.default.createElement(
        _Table2.default,
        TableProps,
        _react2.default.createElement(
          _TableHead2.default,
          TableHeadProps,
          _react2.default.createElement(
            _TableRow2.default,
            TableHeadRowProps,
            columns.map(function (column, index) {
              return _react2.default.createElement(
                _TableCell2.default,
                _extends({
                  key: column.label + '-' + index
                }, TableHeadCellProps),
                _react2.default.createElement(_Renderer.LabelRenderer, { column: column, data: data })
              );
            })
          )
        ),
        _react2.default.createElement(
          _TableBody2.default,
          TableBodyProps,
          data.map(function (row, rowIndex) {
            return _react2.default.createElement(
              _TableRow2.default,
              _extends({ key: rowIndex, className: _this2.getRowClass(rowIndex) }, TableBodyRowProps),
              columns.map(function (column, columnIndex) {
                return _react2.default.createElement(
                  _TableCell2.default,
                  _extends({
                    key: rowIndex + '-' + columnIndex
                  }, TableBodyCellProps),
                  _react2.default.createElement(_Renderer.CellRenderer, { column: column, row: row, data: data })
                );
              })
            );
          })
        ),
        showPagination && _react2.default.createElement(
          _TableFooter2.default,
          null,
          _react2.default.createElement(
            _TableRow2.default,
            null,
            _react2.default.createElement(_Pagination2.default, {
              count: count,
              rowsPerPage: rowsPerPage,
              page: page,
              TablePaginationProps: TablePaginationProps,
              onChangePage: this.handleChangePage
            })
          )
        )
      );
    }
  }]);

  return DataTable;
}(_react.Component);

exports.default = DataTable;
