// material-ui-flat-pagination/lib/PageButton.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Button_1 = require("@material-ui/core/Button");
var styles_1 = require("@material-ui/core/styles");
var classnames_1 = require("classnames");
var core_1 = require("./core");
var styles = function (theme) {
    return styles_1.createStyles({
        root: {
            minWidth: 16
        },
        rootCurrent: {
            paddingLeft: theme.spacing(1.5),
            paddingRight: theme.spacing(1.5)
        },
        rootEllipsis: {
            paddingLeft: theme.spacing(0.5),
            paddingRight: theme.spacing(0.5)
        },
        rootEnd: {
            paddingLeft: theme.spacing(1.5),
            paddingRight: theme.spacing(1.5)
        },
        rootStandard: {
            paddingLeft: theme.spacing(1.5),
            paddingRight: theme.spacing(1.5)
        },
        label: {},
        text: {},
        textPrimary: {},
        textSecondary: {},
        colorInherit: {},
        colorInheritCurrent: {},
        colorInheritOther: {},
        disabled: {},
        sizeSmall: {
            minWidth: 8
        },
        sizeSmallCurrent: {
            paddingLeft: theme.spacing(1),
            paddingRight: theme.spacing(1)
        },
        sizeSmallEllipsis: {
            paddingLeft: theme.spacing(0.25),
            paddingRight: theme.spacing(0.25)
        },
        sizeSmallEnd: {
            paddingLeft: theme.spacing(1),
            paddingRight: theme.spacing(1)
        },
        sizeSmallStandard: {
            paddingLeft: theme.spacing(1),
            paddingRight: theme.spacing(1)
        },
        sizeLarge: {
            minWidth: 24
        },
        sizeLargeCurrent: {
            paddingLeft: theme.spacing(2),
            paddingRight: theme.spacing(2)
        },
        sizeLargeEllipsis: {
            paddingLeft: theme.spacing(0.75),
            paddingRight: theme.spacing(0.75)
        },
        sizeLargeEnd: {
            paddingLeft: theme.spacing(2),
            paddingRight: theme.spacing(2)
        },
        sizeLargeStandard: {
            paddingLeft: theme.spacing(2),
            paddingRight: theme.spacing(2)
        },
        fullWidth: {}
    });
};
var handleClick = function (page, limit, onClick) { return function (ev) {
    onClick(ev, core_1.getOffset(page, limit), page);
}; };
var PageButton = function (props) {
    var _a, _b, _c, _d;
    var limit = props.limit, page = props.page, total = props.total, pageVariant = props.pageVariant, classesProp = props.classes, currentPageColor = props.currentPageColor, disabledProp = props.disabled, disableRippleProp = props.disableRipple, onClickProp = props.onClick, otherPageColor = props.otherPageColor, size = props.size, other = __rest(props, ["limit", "page", "total", "pageVariant", "classes", "currentPageColor", "disabled", "disableRipple", "onClick", "otherPageColor", "size"]);
    var isCurrent = pageVariant === 'current';
    var isEllipsis = pageVariant === 'ellipsis';
    var isEnd = pageVariant === 'end';
    var isStandard = pageVariant === 'standard';
    var isSmall = size === 'small';
    var isLarge = size === 'large';
    var rootCurrent = classesProp.rootCurrent, rootEllipsis = classesProp.rootEllipsis, rootEnd = classesProp.rootEnd, rootStandard = classesProp.rootStandard, colorInheritCurrent = classesProp.colorInheritCurrent, colorInheritOther = classesProp.colorInheritOther, sizeSmallCurrent = classesProp.sizeSmallCurrent, sizeSmallEllipsis = classesProp.sizeSmallEllipsis, sizeSmallEnd = classesProp.sizeSmallEnd, sizeSmallStandard = classesProp.sizeSmallStandard, sizeLargeCurrent = classesProp.sizeLargeCurrent, sizeLargeEllipsis = classesProp.sizeLargeEllipsis, sizeLargeEnd = classesProp.sizeLargeEnd, sizeLargeStandard = classesProp.sizeLargeStandard, classes = __rest(classesProp, ["rootCurrent", "rootEllipsis", "rootEnd", "rootStandard", "colorInheritCurrent", "colorInheritOther", "sizeSmallCurrent", "sizeSmallEllipsis", "sizeSmallEnd", "sizeSmallStandard", "sizeLargeCurrent", "sizeLargeEllipsis", "sizeLargeEnd", "sizeLargeStandard"]);
    classes.root = classnames_1.default(classes.root, (_a = {},
        _a[rootCurrent] = isCurrent,
        _a[rootEllipsis] = isEllipsis,
        _a[rootEnd] = isEnd,
        _a[rootStandard] = isStandard,
        _a));
    classes.colorInherit = classnames_1.default(classes.colorInherit, (_b = {},
        _b[colorInheritCurrent] = isCurrent,
        _b[colorInheritOther] = !isCurrent,
        _b));
    classes.sizeSmall = classnames_1.default(classes.sizeSmall, (_c = {},
        _c[sizeSmallCurrent] = isCurrent && isSmall,
        _c[sizeSmallEllipsis] = isEllipsis && isSmall,
        _c[sizeSmallEnd] = isEnd && isSmall,
        _c[sizeSmallStandard] = isStandard && isSmall,
        _c));
    classes.sizeLarge = classnames_1.default(classes.sizeLarge, (_d = {},
        _d[sizeLargeCurrent] = isCurrent && isLarge,
        _d[sizeLargeEllipsis] = isEllipsis && isLarge,
        _d[sizeLargeEnd] = isEnd && isLarge,
        _d[sizeLargeStandard] = isStandard && isLarge,
        _d));
    var color = isCurrent ? currentPageColor : otherPageColor;
    var disabled = disabledProp || isEllipsis || page <= 0 || total <= 0;
    var disableRipple = disableRippleProp || disabled || isCurrent;
    var onClick;
    if (onClickProp && !disabled && (isEnd || isStandard)) {
        onClick = handleClick(page, limit, onClickProp);
    }
    return (React.createElement(Button_1.default, __assign({ classes: classes, color: color, disabled: disabled, disableRipple: disableRipple, onClick: onClick, size: size }, other)));
};
PageButton.defaultProps = {
    limit: 1,
    page: 0,
    total: 0,
    pageVariant: 'standard',
    disabled: false,
    disableRipple: false
};
var PageButtonWithStyles = styles_1.withStyles(styles, {
    name: 'MuiFlatPageButton'
})(PageButton);
exports.default = PageButtonWithStyles;
